File: Behavioral/Lab14/Typescript/src/Entities/realtor_apartment.ts
import { Apartment } from "./apartment";
import { Customer } from "./customer";
import { RealEstate } from "./real_estate";
import { RealtorAbstract } from "./realtor";

export class RealtorApartment extends RealtorAbstract {
    public override SellEstate(customer: Customer, estate: RealEstate): void {
        if (estate instanceof Apartment) {
            console.log(`${this.nameAndSurname} is selling ${estate.constructor.name} to ${customer.nameAndSurname} with price ${estate.priceInUSD}$`);
        } else if (this._realtor !== null) {
            this._realtor.SellEstate(customer, estate);
        }
    }
}

File: Behavioral/Lab14/C#/Entities/RealtorApartment.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_.Entities
{
    public class RealtorApartment : RealtorAbstract
    {
        public RealtorApartment(string nameAndSurname, string phone, string email, double yearsOfWork) : base(nameAndSurname, phone, email, yearsOfWork)
        {
        }

        public override void SellEstate(Customer customer, RealEstate estate)
        {
            if(estate is Apartment)
                Console.WriteLine($"{NameAndSurname} is selling {estate.GetType().Name} to {customer.NameAndSurname} with price {estate.PriceInUSD}$");
            else if(_realtor != null)
                _realtor.SellEstate(customer, estate);
        }

    }
}
